src/logic/safe/store/actions/processTransaction.ts

```diff
@@ -123,10 +122,14 @@ const processTransaction = ({ approveAndExecute, notifiedTransaction, safeAddres
           await Promise.all([
             saveTxToHistory({ ...txArgs, txHash }),
             storeTx(
-              mockedTx.updateIn(
-                ['ownersWithPendingActions', mockedTx.isCancellationTx ? 'reject' : 'confirm'],
-                (previous) => previous.push(from),
-              ),
+              mockedTx.withMutations((record) => {
```

Comment:
Would be really great if we move this to a separate action (if you have time and desire you can do it in this PR)

Reply:
I tried to do this here, but it's getting ugly, to say the least. I prefer to continue the refactor in a different PR, to not lose the focus of this one.

Reply:
created an issue to work on this: https://github.com/gnosis/safe-react/issues/1342